FlowCrypt/Functionality/Api/Remote Pub Key Apis/PubLookup.swift

```diff
@@ -68,8 +68,13 @@ class PubLookup: PubLookupType {
     }
 
     func fetchRemoteUpdateLocal(with recipient: Recipient) async throws -> RecipientWithSortedPubKeys {
-        let recipient = try await self.lookup(recipient: recipient)
-        try localContactsProvider.updateKeys(for: recipient)
-        return recipient
+        let remoteRecipient = try await self.lookup(recipient: recipient) // This is recipient info from remote
+        if let localContact = try await localContactsProvider.searchRecipient(with: recipient.email), // First check if we have local saved contact
+          localContact.keyState == .revoked, // Check if local contact key is revoked because if local contact key state is revoked, we should not use further keys with same fingerprints. Instead we should use local revoked key from now on
+           localContact.pubKeys.map(\.primaryFingerprint) == remoteRecipient.pubKeys.map(\.primaryFingerprint) { // Even local saved key is revoked key, we should use newer(remote) key if fingerprints(pubkeys) are not same
+           return localContact
+        }
+        try localContactsProvider.updateKeys(for: remoteRecipient)
+        return remoteRecipient
```

Comment:
Ok, this makes a bit more sense but looks brittle.

You can have `try localContactsProvider.updateKeys(for: remoteRecipient)` return the updated set of keys, including local ones. So the signature would be `try localContactsProvider.updateKeys(for: remoteRecipient) -> RecipientWithSortedPubKeys` or similar.

Then the whole method here could be:

```swift
        let remoteRecipient = try await self.lookup(recipient: recipient)
        return try localContactsProvider.updateKeys(for: remoteRecipient)
```

That would make implementation less brittle because you don't have to guess in which situations you should be returning which set of keys. I'm pretty sure there are edge cases in which the current implementation would have incorrect results.

Reply:
Sounds good.